addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_materials.py

```diff
@@ -109,21 +110,53 @@ def __gather_double_sided(blender_material, mesh_double_sided, export_settings):
 
 
 def __gather_emissive_factor(blender_material, export_settings):
-    emissive_socket = gltf2_blender_get.get_socket_or_texture_slot(blender_material, "Emissive")
+    emissive_socket = gltf2_blender_get.get_socket_or_texture_slot_principled(blender_material, "Emission")
+    if emissive_socket is None:
+        emissive_socket = gltf2_blender_get.get_socket_or_texture_slot(blender_material, "Emissive")
     if emissive_socket is None:
         emissive_socket = gltf2_blender_get.get_socket_or_texture_slot_old(blender_material, "EmissiveFactor")
     if isinstance(emissive_socket, bpy.types.NodeSocket):
+
         if emissive_socket.is_linked:
             # In glTF, the default emissiveFactor is all zeros, so if an emission texture is connected,
             # we have to manually set it to all ones.
-            return [1.0, 1.0, 1.0]
+
+            texture_node = __get_tex_from_socket(emissive_socket)
+            if texture_node is None:
+                return [1.0, 1.0, 1.0]
+
+            def is_valid_multiply_node(node):
+                return isinstance(node, bpy.types.ShaderNodeMixRGB) and \
+                    node.blend_type == "MULTIPLY" and \
+                    len(node.inputs) == 3
+
+            multiply_node = next((link.from_node for link in texture_node.path if is_valid_multiply_node(link.from_node)), None)
+            if multiply_node is None:
+                return [1.0, 1.0, 1.0]
+
+            def is_factor_socket(socket):
+                return isinstance(socket, bpy.types.NodeSocketColor) and \
+                    (not socket.is_linked or socket.links[0] not in texture_node.path)
```

Comment:
Since `is_valid_multiply_node` and `is_factor_socket` are both duplicated in `gltf2_blender_gather_materials_pbr_metallic_roughness.py`, could we share and import those functions somewhere?